tools/python/influx-migration/influx_migration.py

```diff
@@ -170,30 +229,44 @@ def bucket_create_rollback(host, token, bucket_name, org, skip_verify):
         client.close()
     return True
 
-def bucket_exists(host, token, bucket_name, skip_verify=False, org=None):
+def bucket_exists(host, token, bucket_name, org_name=None, skip_verify=False):
     """
     Checks for the existence of a bucket.
 
     :param str host: The host for the InfluxDB instance.
     :param str token: The token to use for verification.
     :param str bucket_name: The name of the bucket to verify.
+    :param org_name: The name of the org to use for bucket verification.
+    :type org_name: str or None
     :param bool skip_verify: Whether to skip TLS certificate verification.
-    :param org: The name of the org to use for bucket verification
-    :type org: str or None
     :returns: Whether the bucket exists in the instance.
     :rtype: bool
     """
     try:
-        client = InfluxDBClient(url=host,
-            token=token, timeout=MILLISECOND_TIMEOUT, verify_ssl=not skip_verify, org=org)
-        if client.buckets_api().find_bucket_by_name(bucket_name) is None:
-            return False
+        with InfluxDBClient(url=host, token=token, timeout=MILLISECOND_TIMEOUT,
+            verify_ssl=not skip_verify) as client:
+            # If the org name is provided, set the client org before making
+            # any requests
+            if org_name is not None:
+                client.org = org_name
+            # Buckets may have the same name in multiple organizations
+            buckets = client.buckets_api().find_buckets(name=bucket_name) \
+                if org_name is None else \
+                client.buckets_api().find_buckets(name=bucket_name, org=org_name)
+            if len(buckets.buckets) <= 0:
+                logging.debug(f"Bucket with name {bucket_name} could not be found "
+                    f"in host {host}")
+                return False
+            logging.debug(f"Bucket with name {bucket_name} found in host {host}")
+            if org_name is not None:
```

Comment:
Could we remove this `if`, and instead loop through `buckets.buckets` logging the bucket found and each organization, since at this point we have verified the bucket exists in at least one org.

Reply:
Each bucket's org ID is now logged when any bucket is found.

I also removed exception handling logs in `bucket_exists` since it may be alarming to read "An unexpected error occurred . . ." when the destination bucket is correctly not found.